Makefile

```diff
@@ -1,13 +1,87 @@
-TRIVY_IMAGE_TAG ?= 0.42.1
-OUTPUT_IMAGE ?= mesosphere/trivy-bundles
+TRIVY_VERSION ?= 0.42.1
 
+ROOT_DIR = $(shell cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 TIMESTAMP ?= $(shell date -u +%Y%m%dT%H%M%SZ )
-OUTPUT_IMAGE_TAG = $(TRIVY_IMAGE_TAG)-$(TIMESTAMP)
 
+REGISTRY ?= docker.io
+REPOSITORY ?= mesosphere
+IMAGE ?= trivy-bundles
+TAG ?= $(TRIVY_VERSION)-$(TIMESTAMP)
+
+IMAGE_NAME = $(REPOSITORY)/$(IMAGE):$(TAG)
+IMAGE_NAME_FULL = $(REGISTRY)/$(IMAGE_NAME)
+
+IMAGE_BUNDLE = $(IMAGE)-$(TAG).tar.gz
+
+IMAGES_FILE ?= $(ROOT_DIR)/images.txt
+TAGS_FILE ?= $(ROOT_DIR)/tags.txt
+
+.DEFAULT_GOAL := help
+
+# Enable ONESHELL for all targets
+.ONESHELL:
+
+.PHONY: clean
+clean: ## Clear all intermedieate files
+clean: 
+	rm $(IMAGES_FILE) $(TAGS_FILE)
+
+.PHONY: create-airgapped-image-bundle
+create-airgapped-image-bundle: ## Create airgapped image bundle
+create-airgapped-image-bundle: install-mindthegap latest_image_tag
+	$(call print-target)
+	$(MINDTHEGAP_BIN) create image-bundle --platform linux/amd64 --images-file $(IMAGES_FILE) --output-file $(IMAGE)-`cat $(TAGS_FILE)`.tar.gz
+
+.PHONY: publish-trivy-bundled-image
+publish-trivy-bundled-image: ## Publish image to registry
 publish-trivy-bundled-image: latest_image_tag
-	docker push `cat latest_image_tag`
+	$(call print-target)
+	docker push `cat $(IMAGES_FILE)`
 
 .PHONY: latest_image_tag
+latest_image_tag: ## Build an image with specified version and tag
 latest_image_tag:
-	docker build --build-arg TRIVY_IMAGE_TAG=$(TRIVY_IMAGE_TAG) --build-arg TIMESTAMP=$(TIMESTAMP) -t $(OUTPUT_IMAGE):$(OUTPUT_IMAGE_TAG) .
-	echo $(OUTPUT_IMAGE):$(OUTPUT_IMAGE_TAG) > latest_image_tag
+	$(call print-target)
+	docker build --build-arg TRIVY_IMAGE_TAG=$(TRIVY_VERSION) --build-arg TIMESTAMP=$(TIMESTAMP) -t $(IMAGE_NAME) .
+	echo $(IMAGE_NAME_FULL) > $(IMAGES_FILE)
+	echo $(TAG) > $(TAGS_FILE)
+
+# Tooling needed for mindthegap
+MINDTHEGAP_VERSION ?= v1.11.0
+TOOLS_DIR ?= $(ROOT_DIR)/.local/tools
+
+MINDTHEGAP_BIN = $(TOOLS_DIR)/mindthegap
+
+HOST_ARCH=$(shell uname -m)
+HOST_OS=$(shell uname | tr '[:upper:]' '[:lower:]')
+
+ifeq ($(HOST_ARCH),amd64)
+ARCH := amd64
+else ifeq ($(HOST_ARCH),x86_64)
+ARCH := amd64
+else ifeq ($(HOST_ARCH),arm64)
+ARCH := amd64
+endif
+
+ifeq ($(HOST_OS),linux)
```

Comment:
seems like `OS` and `HOST_OS` is one and the same. can we skip one of the 2?

Reply:
Done.